### PR TITLE
fix list spool file hang problem.

### PR DESCRIPTION
**Pull request by tiantn**
**What It Does**

**How to Test**

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [] added/updated automated tests
- [x] updated the changelog
- [ ] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

File: src/cli/list/spool-files-by-jobid/SpoolFilesByJobid.Handler.ts [hunk 1]
```diff
@@ -23,17 +23,22 @@ export default class ListSpoolFilesByJobidHandler extends FTPBaseHandler {
         this.log.debug("Listing spool files for job ID %s", params.arguments.jobId);
         const job = await JobUtils.findJobByID(params.connection, params.arguments.jobId);
         const files = job.spoolFiles;
+        if (files) {
```
Comment (zFernand0) [inline src/cli/list/spool-files-by-jobid/SpoolFilesByJobid.Handler.ts:26]: Should we check the length as well?
`if (files?.length > 0) { ...`

File: src/cli/list/spool-files-by-jobid/SpoolFilesByJobid.Handler.ts [hunk 2]
```diff
@@ -23,17 +23,22 @@ export default class ListSpoolFilesByJobidHandler extends FTPBaseHandler {
...
-            format: "table",
-        });
+            // Format & print the response
+            params.response.format.output({
+                fields: ["id", "ddName", "procStep", "stepName"],
+                output: files,
+                format: "table",
+            });
+        } else {
+            const failedMessage = params.response.console.log("No spool file.");
```
Comment (zFernand0) [inline src/cli/list/spool-files-by-jobid/SpoolFilesByJobid.Handler.ts:39]: I like how you are making use of the return value from `response.console.log` 🥳

File: src/cli/view/job-status-by-jobid/JobStatusByJobId.Handler.ts [hunk 3]
```diff
@@ -20,7 +20,7 @@ export default class ViewJobStatusByJobIdHandler extends FTPBaseHandler {
         params.response.data.setObj(jobDetails);
         params.response.format.output(
             {
-                fields: ["jobName", "jobId", "owner", "status", "rc", "retcode"],
+                fields: ["jobName", "jobId", "owner", "status", "rc", "retcode", "extra"],
```
Comment (zFernand0) [inline src/cli/view/job-status-by-jobid/JobStatusByJobId.Handler.ts:23]: thanks for adding `extra` as part of the default information shown to the user on `view-job-status` 😋

**Review by zFernand0 — APPROVED** [review]
This looks similar to the following PR:
- https://github.com/zowe/zowe-cli-ftp-plugin/pull/157

Thanks for porting and enhancing the functionality and error message.

Curious if we can link the same issue on the changelog (issue number #156)

**Review by traeok — APPROVED** [review]
LGTM, thanks @tiantn for the fix!

**Review by std4lqi — APPROVED** [review]
LGTM! Thanks, @tiantn !

**Comment by tiantn** [comment]
> This looks similar to the following PR:
> 
>     * [Fix issue #156 to return proper message for active job #157](https://github.com/zowe/zowe-cli-ftp-plugin/pull/157)
> 
> 
> Thanks for porting and enhancing the functionality and error message.
> 
> Curious if we can link the same issue on the changelog (issue number #156)

Hi @zFernand0, I think it fixed the issue #158 😊

**Comment by tiantn** [comment]
Hi @zFernand0 , @traeok , @std4lqi , thank you for your review and good suggestions. I updated the code to check the files length as well. And also update unit test for no spool file scenario to improve test cover rate. Thank you!

**Comment by zFernand0** [comment]
Merging without DCO check. See below issue for details 😋 
- https://github.com/dcoapp/app/issues/211

Kudos to @traeok

**Pull request merged**